app/scripts/modules/core/src/api/api.service.ts

```diff
@@ -12,16 +12,16 @@ interface DefaultParams {
 
 export interface IRequestBuilder {
   config?: IRequestConfig;
-  one?: IRequestBuilder;
-  all?: IRequestBuilder;
-  useCache?: IRequestBuilder;
-  withParams?: IRequestBuilder;
-  data?: IRequestBuilder;
-  get?: IRequestBuilder;
-  getList?: IRequestBuilder;
-  post?: IRequestBuilder;
-  remove?: IRequestBuilder;
-  put?: IRequestBuilder;
+  one?: (...urls: string[]) => IRequestBuilder;
+  all?: (...urls: string[]) => IRequestBuilder;
+  useCache?: (useCache: boolean) => IRequestBuilder;
+  withParams?: (data: any) => IRequestBuilder;
+  data?: (data: any) => IRequestBuilder;
+  get?: (data: any) => IPromise<any>;
+  getList?: (data: any) => IPromise<any>;
+  post?: (data: any) => IPromise<any>;
+  remove?: (data: any) => IPromise<any>;
+  put?: (data: any) => IPromise<any>;
```

Comment:
I like this a lot better

Reply:
Because it's actually correct now?

Reply:
Yeah, I often like correct things better than incorrect things.